bundles/org.openhab.ui/web/build/webpack.config.js

```diff
@@ -236,6 +237,23 @@ module.exports = {
         swSrc: resolvePath('src/service-worker.js')
       })
     ] : []),
+     new CompressionPlugin({
+      filename: '[path][base].gz',
+      algorithm: 'gzip',
+      test: /\.js$|\.css$|\.html$/,
+      threshold: 0,
+      minRatio: Infinity,
+    }),
+    new CompressionPlugin({
+      filename: '[path][base].br',
+      algorithm: 'brotliCompress',
+      test: /\.(js|css|html|svg)$/,
```

Comment:
![image](https://user-images.githubusercontent.com/2004147/195775511-aff6f1a0-2e0c-4bae-9681-f2624345f495.png)

Maybe we could save a few more bytes by compressing additional extensions (.woff, .woff2, .ttf, .mp3, .jpg, .png) though it remains to be seen as they might not have that good a compression ratio.

Otherwise, I think we're good to merge if you don't have anything more to check!

Reply:
Ok, we are good to merge.  Thanks for all the very in-depth testing and advice!   Hopefully this makes a noticeable difference for people.

Reply:
The code above compresses SVG files for brotli, but not for gzip.

The Gzip compression can be improved with 3-8% by compressing with Zopfli: https://webpack.js.org/plugins/compression-webpack-plugin/#using-zopfli instead of the default Gzip implementation.  However all browsers not supporting Brotli were released before 2018, so the GZip compression helps only to browsers released before 2018.

> Maybe we could save a few more bytes by compressing additional extensions (.woff, .woff2, .ttf, .mp3, .jpg, .png) though it remains to be seen as they might not have that good a compression ratio.

JPG is already compressed.  It might be possible to reduce further the jpg file size by stripping meta data, and applying lossless conversions.

SVG files can be made smaller by stripping from then whitespaces and useless elements — https://github.com/openhab/openhab-webui/pull/1430.

PNG files can be reduced by applying to them lossless compressions/optimizations — https://github.com/openhab/openhab-webui/pull/1431 .  My experience is that to reduce the PNG files further one has to convert them to WebP format, instead of applying brotli/gz.